### PR TITLE
refactor: convert PlatformmId to union

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- convert `PlatformmId` to union

## Context
- Ref: [#13743](https://github.com/renovatebot/renovate/issues/13743)

## Documentation (please check one with an [x])
- Maybe we should add it to the docs/best-practices to not use enums.
## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/local/index.ts [hunk 1]
```diff
@@ -9,12 +9,12 @@ import * as gitlab from '../gitlab';
 import type { Preset, PresetConfig } from '../types';
 
 const resolvers = {
-  [PlatformId.Azure]: azure,
-  [PlatformId.Bitbucket]: bitbucket,
-  [PlatformId.BitbucketServer]: bitbucketServer,
-  [PlatformId.Gitea]: gitea,
-  [PlatformId.Github]: github,
-  [PlatformId.Gitlab]: gitlab,
+  ['azure']: azure,
```
Comment (viceice) [inline lib/config/presets/local/index.ts:12]:
```suggestion
  azure: azure,
```
simplify 😉

File: lib/util/http/bitbucket-server.ts [hunk 2]
```diff
@@ -1,16 +1,14 @@
-import { PlatformId } from '../../constants';
 import { resolveBaseUrl } from '../url';
 import type { HttpOptions, HttpResponse, InternalHttpOptions } from './types';
 import { Http } from '.';
-
```
Comment (viceice) [inline lib/util/http/bitbucket-server.ts:5]: please revert this 😉

File: lib/util/http/bitbucket.ts [hunk 3]
```diff
@@ -1,15 +1,13 @@
-import { PlatformId } from '../../constants';
 import type { HttpOptions, HttpResponse, InternalHttpOptions } from './types';
 import { Http } from '.';
-
```
Comment (viceice) [inline lib/util/http/bitbucket.ts:4]: same

**Review by viceice — DISMISSED** [review]

File: lib/config/presets/local/index.ts (hunk 1 above)
Comment (RahulGautamSingh) [reply lib/config/presets/local/index.ts]: TS Error: The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type

Comment (viceice) [reply lib/config/presets/local/index.ts:12]: should probably only be `azure`. you need to change all those assignments

**Review by rarkins — DISMISSED** [review]

**Review by JamieMagee — DISMISSED** [review]

File: lib/config/presets/local/index.ts [hunk 4]
```diff
@@ -9,12 +9,12 @@ import * as gitlab from '../gitlab';
...
 const resolvers = {
-  [PlatformId.Azure]: azure,
-  [PlatformId.Bitbucket]: bitbucket,
-  [PlatformId.BitbucketServer]: bitbucketServer,
-  [PlatformId.Gitea]: gitea,
-  [PlatformId.Github]: github,
-  [PlatformId.Gitlab]: gitlab,
+  azure,
+  bitbucket,
+  ['bitbucket-server']: bitbucketServer,
```
Comment (viceice) [inline lib/config/presets/local/index.ts:14]:
```suggestion
  'bitbucket-server': bitbucketServer,
```
shouldn't this work too? 🤔

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by viceice — DISMISSED** [review]

**Comment by viceice** [comment]
this will cause some conflicts with v33 branch

**Comment by rarkins** [comment]
Let's wait until after then

**Review by viceice — APPROVED** [review]

**Pull request merged**